designer/client/src/components/graph/node-modal/fragment-input-definition/settings/variants/fields/UserDefinedListInput.tsx

```diff
@@ -189,9 +188,9 @@ export const UserDefinedListInput = ({
                         validationLabelInfo={
                             temporaryValuesTyping && <CircularProgress size={"1rem"} sx={(theme) => ({ marginTop: theme.spacing(0.5) })} />
                         }
-                        expressionObj={{ language: ExpressionLang.SpEL, expression: temporaryListItem.trim() }}
+                        expressionObj={{ language: ExpressionLang.SpEL, expression: temporaryListItem }}
                         onValueChange={(value) => {
-                            setTemporaryListItem(value.expression);
+                            setTemporaryListItem(value.expression.trim());
                             setTemporaryValuesTyping(true);
                             setTemporaryValueErrors([]);
                             validateTemporaryListItem(value.expression);
```

Comment:
Shouldn't we also trim this validated value?

Reply:
yup, it makes sense.